lib/puppet/defaults.rb

```diff
@@ -1319,25 +1319,16 @@ def self.initialize_default_settings!(settings)
       by `puppet`, and should only be set if you're writing your own Puppet
       executable.",
     },
-    :serverport => {
-      :default    => 8140,
-      :type       => :port,
-      :desc       => "The default port puppet subcommands use to communicate
-      with Puppet Server. (eg `puppet facts upload`, `puppet agent`). May be
-      overridden by more specific settings (see `ca_port`, `report_port`).",
-      :hook       => proc do |value|
-        Puppet[:masterport] = value unless Puppet.settings.set_by_config?(:masterport)
-      end
-    },
     :masterport => {
       :default    => "$serverport",
```

Comment:
This is adding a circular dependency between `masterport` and `serverport`. Should be changed to `8140` like on 6.x: https://github.com/puppetlabs/puppet/blob/6.x/lib/puppet/defaults.rb#L1375